lib/pipeline.ts

```diff
@@ -241,6 +242,7 @@ Pipeline.prototype.execBuffer = deprecate(function () {
 Pipeline.prototype.exec = function (callback: CallbackFunction) {
   // Wait for the cluster to be connected, since we need nodes information before continuing
   if (this.isCluster && !this.redis.slots.length) {
+    if (this.redis.status === "wait") this.redis.connect();
```

Comment:
```suggestion
    if (this.redis.status === "wait") this.redis.connect().catch(noop);
```

I think that without the Promise.catch, there may be an UnhandledPromiseRejection that would terminate node processes with a `process.on('unhandledRejection')` listener that terminates.

Looks reasonable otherwise, but I'm not a maintainer

Reply:
Added to all three location.